department-of-veterans-affairs/caseflow
# Display end of 90-day submission window for VLJs

Resolves [CASEFLOW-2553](https://vajira.max.gov/browse/CASEFLOW-2553)
Received design approval from Kim via Slack.

### Description
For the header section of Hearing Worksheets, this PR makes three changes:

1. Adds "HEARING" in front of "DATE" (or "HEAR." in the pdf)
2. Moves "HEARING DATE" (or "HEAR. DATE" in the pdf) to a new row
3. Calculates and adds display of "+90 DAYS" and a date, indicating the end of the 90-day evidence submission window. If the date falls on a weekend or holiday, then the end date becomes the next business day.

Note: I found it a bit annoying that I had to put the `weekend_and_holiday` method in both hearing.rb and legacy_hearing.rb in order to quiet reek's FeatureEnvy warnings. Would love to hear if there's a better way to resolve that.

### Testing Plan
1. Switch user to a VLJ user
2. Select a docket from the hearings schedule
3. Select the "Edit VLJ Hearing Worksheet" link on a hearing
4. Verify that the Hearing Date shows on a new row, as in screenshot below
5. Verify that it is labeled "Hearing Date" (rather than "Date")
6. Verify that the end of submission window date appears under the heading "+90 Days" as shown below
7. Click "Save as PDF" and verify in the resulting PDF the same things in steps 4 - 6.


### User Facing Changes
![Screen Shot 2021-12-13 at 2 49 04 PM](https://user-images.githubusercontent.com/92495948/145901608-b0f716e5-9d1d-40f8-ac82-492eb45779fa.png)

![Screen Shot 2021-12-13 at 2 49 18 PM](https://user-images.githubusercontent.com/92495948/145902546-9f196168-2e9f-4885-8055-fcdf65e8e5df.png)
